atest/testdata/running/for.robot

```diff
@@ -317,6 +324,120 @@ For In Range With Expressions Containing Floats
     \    @{result} =    Create List    @{result}    ${i}
     Should Be True    @{result} == [3.14, 1.14, -0.86]
 
+For In Zip
+    @{items}=    Create List    a    b    c    d
+    @{things}=    Create List    e    f    g    h
+    : FOR    ${item}    ${thing}    IN ZIP    ${items}    ${things}
+    \    @{result} =    Create List    @{result}    ${item}:${thing}
+    Should Be True    @{result} == ['a:e', 'b:f', 'c:g', 'd:h']
+
+For In Zip With Uneven Lists
+    [Documentation]    Handling lists with different number of elements.  This will ignore any elements after the shortest list ends, just like with Python's zip()
+    @{items}=    Create List    a    b    c
+    @{things}=    Create List    d    e    f    g    h
+    : FOR    ${item}    ${thing}    IN ZIP    ${items}    ${things}
+    \    @{result} =    Create List    @{result}    ${item}:${thing}
+    Should Be True    @{result} == ['a:d', 'b:e', 'c:f']
+
+For In Zip With 3 Lists
+    [Documentation]    Handling more than one list like :FOR ${a} ${b} ${c} IN ZIP ${x} ${y} ${z}.
```

Comment:
- This doc isn't very useful. 
- Would be nice to test also zipping more than three lists (e.g. 5).
- Tests could be simplified by constructing loops like this:

```robotframework
*** Test Cases ***
Example
    @{lower} =    Evaluate    list('abcde')
    @{upper} =     Evaluate    list('ABCDEFG')
    @{numbers} =    Evaluate    range(100)
    @{expected} =    Create List    a-A-0    b-B-1    c-C-2    d-D-3    e-E-4    not-used-anymore
    :FOR    ${low}    ${up}    ${num}    ${exp}    IN ZIP    ${lower}    ${upper}    ${numbers}    ${expected}
    \    Should Be Equal    ${low}-${up}-${num}    ${exp}
    Should Be Equal    ${exp}    e-E-4
```